app/src/main/java/io/homeassistant/companion/android/sensors/SensorsSettingsFragment.kt

```diff
@@ -184,10 +207,75 @@ class SensorsSettingsFragment : PreferenceFragmentCompat() {
                 prefCategory.addPreference(pref)
             }
         }
+
+        if (showOnlyEnabledSensors) showHideGroupsIfNeeded()
+    }
+
+    private fun getShowOnlyEnabledSensors(showOnlyEnabledSensors: Boolean) {
+        runBlocking {
+            integrationUseCase.setShowOnlyEnabledSensors(showOnlyEnabledSensors)
+        }
+    }
+    private fun getShowOnlyEnabledSensors(): Boolean {
+        var showOnlyEnabledSensors = false
+        runBlocking {
+            showOnlyEnabledSensors = integrationUseCase.getShowOnlyEnabledSensors()
+        }
+        return showOnlyEnabledSensors
+    }
+
+    private fun showHideSensorsIfNeeded(showOnlyEnabledSensors: Boolean) {
+        SensorReceiver.MANAGERS.filter { m -> m.hasSensor(requireContext()) }.forEach { manager ->
+            manager.availableSensors.forEach { sensor ->
+                val pref = findPreference<Preference>(sensor.id)
+                showHidePreference(pref, showOnlyEnabledSensors, manager, sensor.id)
+            }
+        }
+        showHideGroupsIfNeeded()
+    }
+
+    private fun showHidePreference(
+        pref: Preference?,
+        showOnlyEnabledSensors: Boolean,
+        manager: SensorManager,
+        sensorId: String
+    ) {
+        if (pref != null && pref.parent != null) {
+            if (showOnlyEnabledSensors) {
+                pref.isVisible = manager.isEnabled(requireContext(), sensorId)
+            } else {
+                pref.isVisible = true
+            }
+        }
+    }
+
+    private fun getShowOnlyEnabledSensorsSummary(showOnlyEnabledSensors: Boolean): String {
+        return if (showOnlyEnabledSensors) {
+            getString(R.string.only_enabled_sensors_shown)
+        } else {
+            getString(R.string.all_sensors_shown)
+        }
+    }
+
+    private fun showHideGroupsIfNeeded() {
+        for (pref in preferenceScreen) {
+            val prefGroup = pref as PreferenceGroup
+            if (prefGroup != null) {
```

Comment:
As you cast as "pref as PreferenceGroup" it can't be null, It will be exception if pref is not PreferenceGroup, if you want nullable, cast as "pref as? PreferenceGroup"

Reply:
Thanks 👍 Changed that